src/solc.ts

```diff
@@ -124,23 +124,24 @@ const createSolcInput = (solidityFile: string): any => {
  * Shows when there were errors during compilation.
  * @param {Object} compiled - the output object from running solc
  */
-const errorHandling = (compiled: any) => {
+const errorHandling = (compiled: any, toRedecorate: any) => {
   if (!compiled) {
     throw new FilingError(`solc didn't create any output...`);
   } else if (compiled.errors) {
     // something went wrong.
     logger.error(`solc errors:`);
     compiled.errors.map((error: any) => logger.error(error.formattedMessage));
+    var errorDecorator =  toRedecorate.filter((decorator: any) => decorator.charStart === compiled.errors[0].sourceLocation.start);
     throw new FilingError(
-      'Solc Compilation Error: Make sure your .sol contract compiles without Zappify decorators.',
+      `Solc Compilation Error: Make sure your .sol contract compiles without Zappify decorators. Function/ Struct/ Enum named as "${errorDecorator[0].decorator}" at index ${errorDecorator[0].charStart}`
```

Comment:
This looks good, but fails if there is some unrelated error causing solc to fail (because there is no `.decorator` in `errorDecorator`). Could we add an if statement or something so `errorDecorator` is only printed if it's needed?

Reply:
One more thing - sorry! - if a separate error is occurring where we have a decorator, it incorrectly throws an error. e.g. if I add `some uncompilable stuff` just before a correct `known` decorator, the error blames the decorator. Could we make sure solc's error message is the one we expect before we blame the decorator? Something like `expected function name here` and a decorator is present